Upgrade JGit and SBT 1.0.x

oliverlockwood opened the pull request:
I wanted to upgrade the version of JGit, so that we can begin to make use of new features.

In order to do so, as noted in existing PR #121 (which I believe is superseded by this PR), we need to bump the version of the JDK, since JGit no longer supports Java 7.

While I was at it, it seemed like a good opportunity to bump the 1.0.x track of SBT to be `1.0.3` instead of a release candidate.

Could I have a review of this, please?

oliverlockwood commented:
Tagging @dwijnand, @bchazalet, and @hseeberger - sorry to bug you directly, but from looking at the other existing PRs this seems to be a fairly quiet repo, and you were involved in the discussions in #121.

hseeberger commented on test-project/project/build.properties:
```diff
@@ -1 +1 @@
-sbt.version=0.13.7
+sbt.version=0.13.16
```
Why is this still 0.13?

oliverlockwood replied:
Fair point - apologies for missing that.  I'll bump this.

hseeberger commented on README.md:
```diff
@@ -7,13 +7,13 @@ well as allowing other plugins to make use of git.
 Add the following to your `project/plugins.sbt` or `~/.sbt/0.13/plugins/plugins.sbt` file:
 
-    addSbtPlugin("com.typesafe.sbt" % "sbt-git" % "0.9.3")
+    addSbtPlugin("com.typesafe.sbt" % "sbt-git" % "0.10.0")
```
Hm ... these changes should only become effective after a 0.10.0 release, not?

dwijnand replied:
I agree, we should do this on 0.10.0 release: referring to the master version of the README as reference is common.

dwijnand commented on README.md:
```diff
@@ -7,13 +7,13 @@ well as allowing other plugins to make use of git.
 
 ## Installation ##
-This plugin now requires at least **Java 7** (the latest version supporting Java 6 was `0.8.5`)
+This plugin now requires at least **Java 8** (the latest version supporting Java 7 was `0.9.3`)
```
Two things:
* I think it would be useful to preserve the information about the latest version supporting Java 6
* To avoid confusion I think we should rewrite this to avoid saying "now".

So something like:

> As of 0.10.0 (not released yet) this plugin requires at least **Java 8**.
> The latest version supporting Java 7 was `0.9.3`, while the latest version supporting Java 6 was `0.8.5`

hseeberger replied:
+1

oliverlockwood commented:
@dwijnand @hseeberger thank you for the prompt feedback.  I believe I've addressed the comments you raised, and the Travis PR build is green, so please take another look when you have the chance.

dwijnand commented:
I'll just give @hseeberger a chance to weigh in before merging.

hseeberger commented:
LGTM too!

hseeberger commented:
Thanks, @oliverlockwood!

hseeberger commented:
I'll leave a release for @dwijnand ...
